src/SDKs/Azure.Base/data-plane/Azure.Base/Http/HttpPipeline.cs

```diff
@@ -3,40 +3,37 @@
 
 using Azure.Base.Http.Pipeline;
 using System;
-using System.Diagnostics;
+using System.Collections.Generic;
+using System.Reflection;
 using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
+using Azure.Base.Attributes;
 
 namespace Azure.Base.Http
 {
-    public readonly struct HttpPipeline
+    public class HttpPipeline
     {
-        readonly ReadOnlyMemory<HttpPipelinePolicy> _pipeline;
-        readonly IServiceProvider _services;
+        private readonly HttpPipelineTransport _transport;
+        private readonly ReadOnlyMemory<HttpPipelinePolicy> _pipeline;
+        private readonly IServiceProvider _services;
 
         public HttpPipeline(HttpPipelineTransport transport, HttpPipelinePolicy[] policies = null, IServiceProvider services = null)
```

Comment:
Should we consider flipping the order of `services` and `policies` to allow for using `params` with the policies?

Reply:
I don't have a strong opinion here. It's not a customer facing or even client author facing API so I'm not worried about it being very easy to consume.

Reply:
I don't have a strong opinion either, honestly.   An overload or two could take care of the sugar, if we really wanted to make it friendly.  But, if it's something internal to just `Azure.Base` that isn't used for creating client libraries, then I agree.   Totally not worth the effort.